app/services/sessions.service.ts

```diff
@@ -18,144 +18,145 @@ import * as fakeDataServiceModule from "./fake-data.service";
 export class SessionsService {
 
     public sessionsLoaded = false;
+    public ignoreCache = true;
     public items: BehaviorSubject<Array<SessionModel>> = new BehaviorSubject([]);
     private _useHttpService: boolean = false;
     private _allSessions: Array<SessionModel> = [];
     private _searchFilterState: SearchFilterState;
 
     constructor(
-        private _zone: NgZone,
-        private _favoritesService: FavoritesService
+			private _zone: NgZone,
+			private _favoritesService: FavoritesService
     ) { 
-        try {
-            let cahcedSessions = <Array<SessionModel>>JSON.parse(appSettingsModule.getString('ALLSESSIONS', '[]'));
-            if (cahcedSessions.length > 0 ) {
-                this._allSessions = cahcedSessions.map((s) => new SessionModel(s));
-                this.applyCachedFavorites();
-                this.sessionsLoaded = true;
-            }
-        }
-        catch (error) {
-            console.log('Error while retrieveing sessions from the local cache: ' + error);
-        }
+			try {
+				let cahcedSessions = <Array<SessionModel>>JSON.parse(appSettingsModule.getString('ALLSESSIONS', '[]'));
+				if (cahcedSessions.length > 0 ) {
+					this._allSessions = cahcedSessions.map((s) => new SessionModel(s));
+					this.applyCachedFavorites();
+					this.sessionsLoaded = true;
+				}
+			}
+			catch (error) {
+				console.log('Error while retrieveing sessions from the local cache: ' + error);
+			}
     }
 
     public loadSessions<T>(): Promise<T> {
-        return new Promise((resolve, reject) => {
-            if (this.sessionsLoaded) {
-                Promise.resolve(this._allSessions);
-            }
-            else {
-                if (this._useHttpService) {
-                    return this.loadSessionsViaHttp<Array<ISession>>()
-                        .then((newSessions: Array<ISession>) => {
-                            return this.updateSessions<Array<ISession>>(newSessions);
-                        });
-                }
-                else {
-                    return this.loadSessionsViaFaker<Array<ISession>>()
-                        .then((newSessions: Array<ISession>) => {
-                            return this.updateSessions<Array<ISession>>(newSessions);
-                        });
-                }
-            }
-        });
+			return new Promise((resolve, reject) => {
+				if (this.sessionsLoaded && !this.ignoreCache) {
+					Promise.resolve(this._allSessions);
+				}
+				else {
+					if (this._useHttpService) {
+						return this.loadSessionsViaHttp<Array<ISession>>()
+							.then((newSessions: Array<ISession>) => {
+								return this.updateSessions<Array<ISession>>(newSessions);
+							});
+					}
+					else {
+						return this.loadSessionsViaFaker<Array<ISession>>()
+							.then((newSessions: Array<ISession>) => {
+								return this.updateSessions<Array<ISession>>(newSessions);
+							});
+					}
+				}
+			});
```

Comment:
You indentation seems all wonky. Can you fix these?

Reply:
Yeah not sure why VS Code added more tabs. Will fix it.